recipes/opentracing-cpp/all/conanfile.py

```diff
@@ -79,39 +82,50 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "OpenTracing"
         self.cpp_info.names["cmake_find_package_multi"] = "OpenTracing"
 
-        if self.options.shared:
-            self.cpp_info.components["opentracing"].names["cmake_find_package"] = "opentracing"
-            self.cpp_info.components["opentracing"].names["cmake_find_package_multi"] = "opentracing"
-            self.cpp_info.components["opentracing"].libs = ["opentracing"]
-            self.cpp_info.components["opentracing"].bindirs = ["lib"]
-            if self.options.enable_dynamic_load and self.settings.os == "Linux":
-                self.cpp_info.components["opentracing"].system_libs.append("dl")
+        # Default should be bin/
+        self.cpp_info.components["opentracing"].bindirs = ["lib"]
+        self.cpp_info.components["opentracing"].names[
+            "cmake_find_package"
+        ] = "opentracing"
```

Comment:
I think you lost the static by accident in your names https://github.com/opentracing/opentracing-cpp/blob/4bb431f7728eaf383a07e86f9754a5b67575dab0/CMakeLists.txt#L187

```suggestion
        ] = "opentracing" if self.options.shared else "opentracing-static"
```

Space's comment "conan component name" referred to the first bit of it `self.cpp_info.components["opentracing"]` (the ending in the `[]`) the static in the generatate component name was very much correct

Reply:
I removed the `static` to simplify it since we can not build shared and static libs at the same time.
But now I realize that this will break compatibility with `CMakeLists` using the cmake targets, right?